File: c/problem_90.c
#include "problem_90.h"
#include <stdio.h>
#include <stdlib.h>
#include <assert.h>

int random_excluding(int n, int *exclude, int excl_size) {
    int valid[64], vn=0;
    for(int i=0;i<n;i++){
        int excl=0;
        for(int j=0;j<excl_size;j++) if(exclude[j]==i){excl=1;break;}
        if(!excl) valid[vn++]=i;
    }
    return valid[rand()%vn];
}

void problem90(void) {
    printf("Problem 90:\n");
    int excl[]={0,1,2,4};
    for(int i=0;i<100;i++) assert(random_excluding(5,excl,4)==3);
    int counts[5]={0};
    int excl2[]={2};
    for(int i=0;i<1000;i++) counts[random_excluding(5,excl2,1)]++;
    for(int k=0;k<5;k++) if(k!=2) assert(counts[k]>100);
    printf("All Tests Passed\n");
}

File: rust/problem_90.rs
use rand::Rng;
use std::collections::HashSet;

pub fn random_excluding(n: usize, exclude: &[usize]) -> usize {
    let excl: HashSet<usize> = exclude.iter().cloned().collect();
    let valid: Vec<usize> = (0..n).filter(|i| !excl.contains(i)).collect();
    valid[rand::thread_rng().gen_range(0..valid.len())]
}

pub fn problem90() {
    println!("Problem 90:");
    for _ in 0..100 { assert_eq!(random_excluding(5, &[0,1,2,4]), 3); }
    let mut counts = [0u32; 5];
    for _ in 0..1000 { counts[random_excluding(5, &[2])] += 1; }
    for &k in &[0,1,3,4] { assert!(counts[k] > 100); }
    println!("All Tests Passed");
}
